nexus/types/src/deployment/planning_input.rs

```diff
@@ -496,12 +499,23 @@ impl SledResources {
         &self,
         filter: DiskFilter,
     ) -> impl Iterator<Item = (&ZpoolUuid, &SledDisk)> + '_ {
-        self.zpools.iter().filter_map(move |(zpool, disk)| {
+        self.zpools.iter().filter_map(move |(zpool, (disk, _datasets))| {
             filter
                 .matches_policy_and_state(disk.policy, disk.state)
                 .then_some((zpool, disk))
         })
     }
+
+    pub fn all_datasets(
+        &self,
+        filter: ZpoolFilter,
+    ) -> impl Iterator<Item = (&ZpoolUuid, &Vec<DatasetConfig>)> + '_ {
```

Comment:
Updated (I know this matter more in the argument path, do you think it matters as much for an output parameter?)

Reply:
Not particularly; it just jumped out at me because `&Vec` is kinda weird (I don't think it really gives anything that `&[]` doesn't?). Maaaaybe there's an argument for `&Vec` in a public API making it harder to switch the internal storage to some other type that can also expose a slice?